> /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/test/NameBinding/name_lookup.swift:615:41: error: unexpected error produced: property 'x' references itself
01:16:43 struct PatternBindingWithTwoVars1 { var x = 3, y = x }

Weird... I have no idea how my changes could possibly break this. Let me rebase and try again.

--- a/lib/AST/DeclContext.cpp
+++ b/lib/AST/DeclContext.cpp
@@ -1039,8 +1039,6 @@ bool DeclContext::hasValueSemantics() const {
   if (getExtendedProtocolDecl())
     return !isClassConstrainedProtocolExtension();
   return !getDeclaredInterfaceType()->hasReferenceSemantics();
-
-  return false;
 }
 
 bool DeclContext::isClassConstrainedProtocolExtension() const {

--- a/test/decl/ext/extensions.swift
+++ b/test/decl/ext/extensions.swift
@@ -157,7 +157,7 @@ extension DoesNotImposeClassReq_1 where Self: JustAClass {
     wrappingProperty3 = "" // Okay
   }
   
-  func bar() { // expected-note {{mark method 'mutating' to make 'self' mutable}}{{2-2=mutating }}
+  func bar() { // expected-note {{mark method 'mutating' to make 'self' mutable}}{{3-3=mutating }}
     property = "" // Okay
     wrappingProperty1 = "" // Okay
     wrappingProperty2 = "" // Okay
@@ -178,6 +178,7 @@ instanceOfJustAClass1.wrappingProperty2 = "" // Okay
 instanceOfJustAClass1.wrappingProperty3 = "" // expected-error {{cannot assign to property: 'instanceOfJustAClass1' is a 'let' constant}}
 instanceOfJustAClass1.foo() // expected-error {{cannot use mutating member on immutable value: 'instanceOfJustAClass1' is a 'let' constant}}
 instanceOfJustAClass1.bar() // Okay
+instanceOfJustAClass1.baz() // Okay
 
 var instanceOfJustAClass2 = JustAClass()
 instanceOfJustAClass2.foo() // Okay
@@ -204,22 +205,22 @@ extension DoesNotImposeClassReq_2 where Self : AnyObject {
   
   mutating func foo() {
     property = "" // Okay
-    wrappingProperty1 = "" // Okay
-    wrappingProperty2 = "" // Okay
+    wrappingProperty1 = "" // Okay (the error is on the setter declaration above)
+    wrappingProperty2 = "" // Okay (the error is on the setter declaration above)
     wrappingProperty3 = "" // Okay
   }
   
-  func bar() { // expected-note 2{{mark method 'mutating' to make 'self' mutable}}{{2-2=mutating }}
+  func bar() { // expected-note 2{{mark method 'mutating' to make 'self' mutable}}{{3-3=mutating }}
     property = "" // expected-error {{cannot assign to property: 'self' is immutable}}
-    wrappingProperty1 = "" // Okay
-    wrappingProperty2 = "" // Okay
+    wrappingProperty1 = "" // Okay (the error is on the setter declaration above)
+    wrappingProperty2 = "" // Okay (the error is on the setter declaration above)
     wrappingProperty3 = "" // expected-error {{cannot assign to property: 'self' is immutable}}
   }
   
   nonmutating func baz() { // expected-note 2{{mark method 'mutating' to make 'self' mutable}}
     property = "" // expected-error {{cannot assign to property: 'self' is immutable}}
-    wrappingProperty1 = "" // Okay
-    wrappingProperty2 = "" // Okay
+    wrappingProperty1 = "" // Okay (the error is on the setter declaration above)
+    wrappingProperty2 = "" // Okay (the error is on the setter declaration above)
     wrappingProperty3 = "" // expected-error {{cannot assign to property: 'self' is immutable}}
   }
 }

--- a/test/decl/ext/extensions.swift
+++ b/test/decl/ext/extensions.swift
@@ -164,7 +164,7 @@ extension DoesNotImposeClassReq_1 where Self: JustAClass {
     wrappingProperty3 = "" // expected-error {{cannot assign to property: 'self' is immutable}}
   }
   
-  nonmutating func baz() { // expected-note {{mark method 'mutating' to make 'self' mutable}}
+  nonmutating func baz() { // expected-note {{mark method 'mutating' to make 'self' mutable}}{{3-14=mutating}}
     property = "" // Okay
     wrappingProperty1 = "" // Okay
     wrappingProperty2 = "" // Okay
@@ -191,11 +191,13 @@ extension DoesNotImposeClassReq_2 where Self : AnyObject {
   var wrappingProperty1: String {
     get { property }
     set { property = newValue } // expected-error {{cannot assign to property: 'self' is immutable}}
+    // expected-note@-1 {{mark accessor 'mutating' to make 'self' mutable}}{{5-5=mutating }}
   }
   
   var wrappingProperty2: String {
     get { property }
     nonmutating set { property = newValue } // expected-error {{cannot assign to property: 'self' is immutable}}
+    // expected-note@-1 {{mark accessor 'mutating' to make 'self' mutable}}{{5-16=mutating}}
   }
   
   var wrappingProperty3: String {
@@ -217,7 +219,7 @@ extension DoesNotImposeClassReq_2 where Self : AnyObject {
     wrappingProperty3 = "" // expected-error {{cannot assign to property: 'self' is immutable}}
   }
   
-  nonmutating func baz() { // expected-note 2{{mark method 'mutating' to make 'self' mutable}}
+  nonmutating func baz() { // expected-note 2{{mark method 'mutating' to make 'self' mutable}}{{3-14=mutating}}
     property = "" // expected-error {{cannot assign to property: 'self' is immutable}}
     wrappingProperty1 = "" // Okay (the error is on the setter declaration above)
     wrappingProperty2 = "" // Okay (the error is on the setter declaration above)

